### PR TITLE
[test] Use an in-repo font for test-multithread

### PR DESCRIPTION
**Pull request by ebraminio**

### CODE REVIEW

**Comment by behdad** [comment]
This wouldn't work in out-of-tree builds.  See other tests that use `g_test_build_filename`.

**Comment by behdad** [comment]
Also should check what asan fail is about.

**Comment by behdad** [comment]
Also, you probably should change this line:
```
  hb_buffer_add_utf8 (buffer, text, sizeof (text), 0, sizeof (text)); 
```
to
```
  hb_buffer_add_utf8 (buffer, text, sizeof (text), 0, -1); 
```

File: test/api/hb-subset-test.h [hunk 1]
```diff
@@ -57,6 +57,11 @@ hb_subset_test_open_font (const char *font_path)
 #endif
 
   hb_blob_t* blob = hb_blob_create_from_file (path);
+  if (hb_blob_get_length (blob) == 0)
+  {
+    printf ("The test font is not found.");
+    exit (1);
+  }
```
Comment (ebraminio) [inline test/api/hb-subset-test.h]: Adding the check here, does it look good?

Comment (behdad) [reply test/api/hb-subset-test.h]: Not sure what you mean.

File: test/fuzzing/main.cc [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
 
 int main(int argc, char **argv) {
   hb_blob_t *blob = hb_blob_create_from_file (argv[1]);
   unsigned int len;
   const char *font_data = hb_blob_get_data (blob, &len);
+  if (len == 0)
+  {
+    printf ("The test font is not found.");
+    exit (1);
+  }
```
Comment (ebraminio) [inline test/fuzzing/main.cc]: And here.

File: test/api/hb-subset-test.h (hunk 1 above)
Comment (ebraminio) [reply test/api/hb-subset-test.h]: The same as https://github.com/harfbuzz/harfbuzz/commit/f72b748371da2d7ce327a5d15feea46960aa7dd3#diff-7e54590da7a9eb1143ac164a13c1f300L9

File: test/api/test-multithread.c [hunk 3]
```diff
@@ -135,16 +127,34 @@ test_body (void)
 int
 main (int argc, char **argv)
 {
+  g_test_init (&argc, &argv, NULL);
```
Comment (ebraminio) [inline test/api/test-multithread.c]: Is this look good?

**Comment by behdad** [comment]
Just do whatever is needed :)

**Comment by behdad** [comment]
I'm trying to figure out the asan issue.  It's wrong to pass sizeof(text) to buffer API, but it shouldn't result in bad memory access.

**Comment by behdad** [comment]
> I'm trying to figure out the asan issue. It's wrong to pass sizeof(text) to buffer API, but it shouldn't result in bad memory access.

Ah, text is a pointer, not array.  Makes sense.  Using -1 is correct fix.

**Pull request merged**